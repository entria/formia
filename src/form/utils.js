import get from 'lodash.get';
import set from 'lodash.set';
import isEqual from 'lodash.isequal';

export const handleInitialValuesUpdate = (
  currentValues = {},
  currentInitialValues = {},
  newInitialValues = {},
) => {
  const updatedValues = { ...currentValues };

  Object.keys(newInitialValues).map(fieldName => {
    const fieldCurrentValue = get(currentValues, fieldName, null);
    const fieldCurrentInitialValue = get(currentInitialValues, fieldName, null);
    const isFieldPristine = isEqual(fieldCurrentValue, fieldCurrentInitialValue);

    if (!isFieldPristine) {
      return;
    }

    const fieldNewInitialValue = get(newInitialValues, fieldName, null);
    set(updatedValues, fieldName, fieldNewInitialValue);
  });

  return updatedValues;
};
