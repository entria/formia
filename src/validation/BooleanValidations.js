// @flow
import { Strings } from '@entria/utils';

import type { RuleError } from './Validation';

export const isTrue = () => (value: any): ?RuleError => {
  if (Strings.isEmpty(value)) {
    return null;
  }

  const isValid = value === true;
  if (isValid) {
    return null;
  }

  return {
    code: 'Validation.Booleans.isTrue',
    message: 'Must be true',
    params: [],
  };
};

export const isFalse = () => (value: any): ?RuleError => {
  if (Strings.isEmpty(value)) {
    return null;
  }

  const isValid = value === false;
  if (isValid) {
    return null;
  }

  return {
    code: 'Validation.Booleans.isFalse',
    message: 'Must be false',
    params: [],
  };
};
