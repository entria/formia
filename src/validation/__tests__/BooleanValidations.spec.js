import * as Booleans from '../BooleanValidations';

import { setupTest } from '../../../test/utils';

beforeEach(async () => setupTest());

it('isTrue - empty', () => {
  expect(Booleans.isTrue()()).toMatchSnapshot();
});

it('isTrue - true', () => {
  expect(Booleans.isTrue()(true)).toMatchSnapshot();
});

it('isTrue - another value', () => {
  expect(Booleans.isTrue()('anything here')).toMatchSnapshot();
});

it('isFalse - empty', () => {
  expect(Booleans.isFalse()()).toMatchSnapshot();
});

it('isFalse - false', () => {
  expect(Booleans.isFalse()(false)).toMatchSnapshot();
});

it('isFalse - another value', () => {
  expect(Booleans.isFalse()('anything here')).toMatchSnapshot();
});
