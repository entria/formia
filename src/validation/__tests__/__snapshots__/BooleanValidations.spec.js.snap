// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`isFalse - another value 1`] = `
Object {
  "code": "Validation.Booleans.isFalse",
  "message": "Must be false",
  "params": Array [],
}
`;

exports[`isFalse - empty 1`] = `null`;

exports[`isFalse - false 1`] = `null`;

exports[`isTrue - another value 1`] = `
Object {
  "code": "Validation.Booleans.isTrue",
  "message": "Must be true",
  "params": Array [],
}
`;

exports[`isTrue - empty 1`] = `null`;

exports[`isTrue - true 1`] = `null`;
